docs/DeveloperGuide.md

```diff
@@ -1,38 +1,323 @@
+|  v1.0   |  business owner with outdated products   |           remove products from my database           | indicate that my business is not selling these products anymore |
+|  v1.0   |         forgetful business owner         |           see all my products in one list            |            find out what products my business sells             |
+|  v1.0   |          curious business owner          |       the data of the products in my database        |       find out more about the performance of that product       |
+|  v1.0   |              business owner              |     modify the data of a product in my database      |        keep the information about my product up-to-date         |
+|  v1.0   |    business owner with many products     |         search for a product in my database          |         find out if that product is sold by my business         |
+|  v1.0   |        for-profit business owner         |                    track my sales                    |            manage my business in a profitable manner            |
+|  v2.0   |         forgetful business owner         |     have a list of possible commands and formats     |                 recall what commands I can use                  |
+|  v2.0   |              business owner              |                   save my database                   |   reuse the existing database when I restart the application    |
+
 
 ## Non-Functional Requirements
+* Should work on any _mainstream OS_ with Java 17 installed.
+* Should serve a business with up to 10,000 goods for sale.
+* A user with decent typing speed for normal text should be able to complete most tasks faster through typing
+out commands, compared to using the mouse to navigate a GUI application.
 
-{Give non-functional requirements}
 
 ## Glossary
+<!-- @@author amirhusaini06 -->
+* **Business Credentials** - Unique identifiers (ID, name, password) for a registered business.
+* **Command Parser** - A component responsible for interpreting user commands and triggering appropriate actions.
+* **Inventory Manager** - Manages stock, allowing addition, deletion, and updates of products.
+* **Sales Manager** - Tracks product sales and ensures accurate revenue reporting.
+* **Revenue Calculator** - Computes total revenue and generates sales reports.
+* **CLI (Command-Line Interface)** - A text-based interface where users interact with the application. 
+<!-- @@author b1inmeister -->
+* **Mainstream OS** - Windows, Linux, Unix, macOS 
+
+
+## Instructions for Manual Testing
+### Launch
+1. Download the `.jar` file and copy into an empty folder.
+2. Open Command Prompt on Windows / Terminal on macOS.
+3. Change the current working directory to the folder containing the `.jar` file.
+4. run `java -jar <file name>.jar` on Command Prompt / Terminal.
+
+### Adding a product
```

Comment:
Not tooo sure but the forward slashes in this section seem to be wrongly used. You might want to check if the markdown you have written needs to be corrected.